### ISSUE TITLE
Ploughs 

### ISSUE THREAD

**Issue by pops64** [Courseplay/courseplay#1406]
The code that inverts the tool off set(switches from right to left) is still enabled even though plough rotation is disabled(Assuming your working on re-enabling :D). Also the Lemken Titan 11 is trying to do 3 point turns.(I think it is not being detected as wheeled implent) I do not know if the Salford 8312 does the same. Thanks for getting out a beta and keep up the good work.

**Comment by knoxville73** [comment]
The plow does not turn (tested with Lemken, Kuhn and Amazone plow)

**Comment by knoxville73** [comment]
Some trained tools are linked at the 3 points system (Lemken Titan, Amazone Condor, ...).  I think i'ts a problem for the recognisation as trained tools !

**Comment by Satissis** [comment]
Try 010 and report back if that fixed it :)

**Comment by knoxville73** [comment]
The rotation of the plow is now correct, but we have always the problem for the manuver with the Lemken Titan.

**Comment by Satissis** [comment]
Yeah I know of that one. Something I'll have to look into.

**Comment by pops64** [comment]
So Tipper offeset works, plough rotation works. A few bugs that I ran into.

1) The Salford 8312 turns fine but if you leave turn on field enabled it still tries to reverse this plough. Also if you stop this plough during a turn maneuver you have to lower it manually before resuming course. [Debug 17](http://pastebin.com/WXD5mbLF). 

2) You already know about the lemken. 

3) Also the turn radius for the Salford 4204 is way to high. A hard code turn radius for this plough would be in order cause of the way the plough is designed. (Mostly Ridged tractor joint but the wheel on the plough turns 360 So you can go down to tractor turn radius for this plough)

Not really a bug but a suggestion if you get time. Would it be possible to go the back maker for the plough on lane finish. And front marker for plough on lane start.(To get rid of the triangle that devolpes, though this would be bad on field creation. Mabye an option in the hud for this behavior)

Thanks for working on this

**Comment by gec100** [comment]
I can add that Discbine 313 mower has a same problem as Lemken plow.

**Comment by knoxville73** [comment]
The courseplay is not adapted for the Salford.  This model of plow do not use the same way of others plow. 
See below the correct way for the salford type
![circuit](https://cloud.githubusercontent.com/assets/23388102/20846693/249a7032-b8cb-11e6-82c4-b2baf941432d.jpg)

**Comment by Satissis** [comment]
@knoxville73 that's very true, but CP is just not build for that type. it would require an rewrite of the generator to get that to work :) I'm also guessing that you are only pointing it out ;)

Btw: there is also an 3th way to use it, which is splitting the field into 2 in the middle. then in one direction, you drive at the field edge and the opposite direction you drive in the middle of the field, if you know what I mean :)

**Comment by knoxville73** [comment]
Yes, it's just an info.  I don't use this plow.
I prefer the others plow (Lemken, Amazone, Kuhn)

**Comment by Satissis** [comment]
Try v.024 or later and report back.

I have tested all ploughs now and fine tuned them all.
The Lemken Titan is offcenter from the tractor, so we cant reverse with it. I have also added in an fixed turn radius for it, so it will make tighter turns.
Salford 8312 do not reverse anymore.
Salford 4204 turn radius are also fixed now. just remember if the tractors turn radius is bigger than the plough, it will use the tractors turn radius.

**Comment by pops64** [comment]
It works great. Thanks for the update. Two things but these can be fixed when you get the time. the Salford 8312 might be able to be set to 0.5 m instead of 0.2 m. More play testing required for that. Second if you stop the Salford 8312 in the middle of a turn maneuver and then resume it CP wont lower the plough it must be done manually. And then my suggestion of where the plough stops when it goes to turn(Its not noticeable on the small ploughs but the Salford 8312 and lemken you leave behind a pretty big triangle). But it works which is all that is important. Thanks again, and as always keep up the great work

**Comment by Satissis** [comment]
I'll have an look at it, since once that is done, I should not have to look into the ploughs anymore and can concentrate on other stuff that need fixing and fine tuning :P

**Comment by Satissis** [comment]
Test again with v.025 or later and report back.
Salford 8312 auto tool offset changed to 0.5m

**Comment by pops64** [comment]
Works great. Thanks for update. All things are good to go. Just my suggestions about ploughs, but do you want me to create another thread for that, or  do have it on your pivotal tracker and we can close this out? Thanks again

**Comment by Satissis** [comment]
I would suggest make an new issue about that, thou we don't look through suggestions atm as the same with mods :)

Will close this one now since it's fixed

**Comment by Ian1357** [comment]
Fairly new to courseplay have just installed the developers edition. When I set the ploughing option to do the headland second. It goes along the field then stops and says finished on the first path. Can anyone help?

**Comment by pops64** [comment]
You probably are starting CP on the headland waypoints. Make sure you are using first waypoint option or that CP starts at a low numbered waypoint. Follow http://courseplay.github.io/courseplay/en/index.html it is about 80% accurate and up to date but the basics haven't really changed.

**Comment by Ian1357** [comment]
Thank you for such a quick reply I will try this later and let you know.
Cheers
Ian
